cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -222,6 +224,58 @@ func (ks *kube2sky) handleEndpointAdd(obj interface{}) {
 	}
 }
 
+func (ks *kube2sky) handlePodCreate(obj interface{}) {
+	if e, ok := obj.(*kapi.Pod); ok {
+		//If the pod ip is not yet available, do not attempt to create
+		if e.Status.PodIP != "" {
+			name := buildDNSNameString(ks.domain, podSubdomain, e.Namespace, santizeIP(e.Status.PodIP))
+			ks.mutateEtcdOrDie(func() error { return ks.generateRecordsForPod(name, e) })
+		}
+	}
+}
+
+func (ks *kube2sky) handlePodUpdate(old interface{}, new interface{}) {
+	oldPod, okOld := old.(*kapi.Pod)
+	newPod, okNew := new.(*kapi.Pod)
+
+	// validate that the objects are good and that we have ip's
+	if okOld && okNew && oldPod.Status.PodIP != "" && newPod.Status.PodIP != "" {
+		if oldPod.Status.PodIP != newPod.Status.PodIP {
```

Comment:
what if the oldPod had an IP and the new Pod does not have an IP ?
we should still delete the oldPod entry

can you also make calls to add/deletePod from update so that your code is reused